### PR TITLE
Add developer getting started docs page

### PR DESCRIPTION
**Pull request by cdeil**
This pull request adds a "Gammapy development getting started" page to the Gammapy developer documentation section.

@adonath @joleroi - Does the outline roughly look OK to you?

Personally I'm not using conda or virtual environments for each branch when working on Gammapy, I just constantly overwrite one install. I know this isn't great, but I think maybe good enough for now?
Do you use one environment per branch? Do you think it would be better to explain this more complicated but also better workflow to new developers?

### CODE REVIEW

**Comment by adonath** [comment]
I do it the same, I only use one install across several branches. I have two (not virtual) environments for Python 2.7 and Python 3.5 using two different anaconda installations.

File: docs/development/intro.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+------------
+
+The first step should be to get in touch via the Gammapy mailing list or a Github issue, and to
+let us know what you want to add. This will spare you the frustration of developing something
+that doesn't fit somehow and has to be re-worked, or is already in Gammapy in a different form
+or is not appropriate for inclusion at all.
+
+Get set up
+----------
+
```
Comment (adonath) [inline docs/development/intro.rst]: Maybe still add a link to http://docs.gammapy.org/en/latest/install.html at the beginning? To explain how to setup a an Anacaonda Python installation...

**Comment by joleroi** [comment]
I also don't have a separate conda environment for each branch. I just have some environments with different version of sherpa, astropy etc. and sometimes use them for testing.

File: docs/development/intro.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+* explain make and setup.py
+
+Write a working example
+-----------------------
+
+* Explain "documentation driven development" and "test driven development"
+
+* make a branch
+* test in `examples`
+* `import IPython; IPython.embed` trick
```
Comment (joleroi) [inline docs/development/intro.rst]: I would also point people to this https://docs.python.org/3/library/pdb.html#module-pdb. I find it more useful for debugging than ``embed`` (cycle to stack trace)

**Comment by cdeil** [comment]
I've worked on this a bit more today:

- Added  a page `development/setup.rst` that describes the Gammapy technical project "setup", i.e. the repositories and services we use.
- Added some info in `development/intro.rst`, but that is very unfinished.
- Remove old and broken `gammapy-conda-install.sh`,  add `environment-dev.yml`.
  This is also unfinished, but it's a start.

I plan to add another `gammapy-extra/environment.yml` now, which is supposed to contain all dependencies that are needed to execute the tutorial notebooks. This will also be suitable for most users to just have an environment that can be used for any Gammapy analysis. But it will not solve the problem yet of having "known good environments" with stable versions of Gammapy and it's dependencies, which is what we probably should provide with each stable release, as something easy to install and use to produce "production" results e.g. for papers or generally for results shared with colleagues.

@adonath and I plan to continue to improve Gammapy installation setup and documentation a bit in the coming days, this is not ready for review. But if anyone has feedback or suggestions, or wants to help, please comment here or in new issues or pull requests!

Merging this now ...

**Pull request merged**